### PR TITLE
#9178 Document/Refactor steps for dispatch_event in eventdispatcher.rs

### LINKED ISSUE #9178: Document/Refactor steps for dispatch_event in eventdispatcher.rs
The spec may have changed after it was last modified, so refactoring may be necessary

Code: `components/script/dom/eventdispatcher.rs`
Spec: https://dom.spec.whatwg.org/#concept-event-dispatch

### PR DESCRIPTION
**Pull request by iszak**
This fixes https://github.com/servo/servo/issues/9178

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9409)

### CODE REVIEW

File: components/script/dom/eventdispatcher.rs [hunk 1]
```diff
@@ -137,17 +148,18 @@ pub fn dispatch_event(target: &EventTarget, pseudo_target: Option<&EventTarget>,
...
         Some(pseudo_target) => pseudo_target,
         None => target.clone(),
     });
 
-    if event.stop_propagation() {
-        return !event.DefaultPrevented();
-    }
-
-    event.set_dispatching(true);
-
```
Comment (KiChjang) [inline components/script/dom/eventdispatcher.rs]: Why is this removed?

File: components/script/dom/eventdispatcher.rs [hunk 2]
```diff
@@ -105,14 +110,17 @@ fn dispatch_to_listeners(event: &Event, target: &EventTarget, chain: &[&EventTar
     assert!(!event.stop_propagation());
     assert!(!event.stop_immediate());
 
-    /* bubbling */
+    // 8. If event's bubbles attribute value is true, run these substeps:
     if !event.bubbles() {
         return;
     }
 
-    event.set_phase(EventPhase::Bubbling);
```
Comment (KiChjang) [inline components/script/dom/eventdispatcher.rs]: Why is this removed?

Comment (iszak) [inline components/script/dom/eventdispatcher.rs]: I moved this after the next block (see the comment for 8.2) as I felt this reflected the spec

File: components/script/dom/eventdispatcher.rs (hunk 1 above)
Comment (iszak) [inline components/script/dom/eventdispatcher.rs]: I think the removal of event.stop_propagation() is a mistake since it will fail the assert! in dispatch_to_listeners() so I will re-instate it - but I'm not sure where in the step of events it belongs.

As for the set_dispatching, this is moved here https://github.com/servo/servo/pull/9409/files#diff-e70183cc533bf121ea8a9c077ea36d90R152 since it's the first step

Comment (KiChjang) [inline components/script/dom/eventdispatcher.rs]: cc @Ms2ger

**Comment by Ms2ger** [comment]
I don't think it's useful to have full spec quotes in general; we use "Step 4." instead.

**Comment by iszak** [comment]
@Ms2ger I think we should keep them for the following reason;

- If the specification changes (as it has done before), this will document the intended steps - an alternative to this is use a versioned url to the referenced specification of which the code was implemented
- Although the code in my opinion is easy to follow, I do think the comments add the ability to skip over blocks without attempting to understand it.

Although I admit it does have some down sides
- Having to potentially re-align the wording with the specification (not specifically the order of execution as is this ticket)
- Can be seen to add some noise to the code

Since this is a minor task that shouldn't consume much time, if you still want to shorten them let me know.

More importantly are you able to feedback on the cc'ed by @KiChjang

File: components/script/dom/eventdispatcher.rs [hunk 3]
```diff
@@ -105,14 +109,17 @@ fn dispatch_to_listeners(event: &Event, target: &EventTarget, chain: &[&EventTar
     assert!(!event.stop_propagation());
     assert!(!event.stop_immediate());
 
-    /* bubbling */
+    // Step 8.
     if !event.bubbles() {
         return;
     }
```
Comment (KiChjang) [inline components/script/dom/eventdispatcher.rs]: Step 8.1 is missing.

Comment (iszak) [inline components/script/dom/eventdispatcher.rs]: The comment is missing, but I believe it's still being done when we build the chain it's in tree order ( https://dom.spec.whatwg.org/#concept-tree-order ) here https://github.com/servo/servo/pull/9409/files#diff-e70183cc533bf121ea8a9c077ea36d90R159 and when we do the capturing phase it's in reverse tree order https://github.com/servo/servo/pull/9409/files#diff-e70183cc533bf121ea8a9c077ea36d90L68. I am not sure the best way to document this.

**Comment by nox** [comment]
I was looking into this to rebase it and ended up introducing new functions and whatnot in #9787. Thanks for your work!
